interp/cfg.go

```diff
@@ -2302,6 +2305,20 @@ func (interp *Interpreter) cfg(root *node, sc *scope, importPath, pkgName string
 	return initNodes, err
 }
 
+// fixUntyped propagates implicit type conversions for untyped binary expressions.
+func fixUntyped(nod *node, sc *scope) {
+	nod.Walk(func(n *node) bool {
+		if n == nod || (n.kind != binaryExpr && n.kind != parenExpr) || !n.typ.untyped {
+			return true
+		}
+		n.typ = nod.typ
+		if n.findex >= 0 {
+			sc.types[n.findex] = nod.typ.frameType()
+		}
+		return true
```

Comment:
Does it ever implicitly return false?

Reply:
A callback returning false terminates the tree walk. In our case, we just want to skip the node, but continue to process the rest of the expression subtree, looking for untyped expressions, so the callback always returns true.